auth-foundation/src/main/java/com/okta/authfoundation/client/AccessTokenValidator.kt

```diff
@@ -31,20 +31,20 @@ fun interface AccessTokenValidator {
     class Error(message: String) : IllegalStateException(message)
 
     /**
-     * Called when the [OidcClient] receives a [Token] response.
+     * Called when the [OAuth2Client] receives a [Token] response.
      *
      * This should throw an [Exception] if the token is invalid.
      *
-     * @param oidcClient the [OidcClient] that made the [Token] request.
+     * @param OAuth2Client the [OAuth2Client] that made the [Token] request.
      * @param accessToken the access token from the [Token] response.
      * @param idToken the [Jwt] representing the id token from the [Token] response.
      */
-    suspend fun validate(oidcClient: OidcClient, accessToken: String, idToken: Jwt)
+    suspend fun validate(OAuth2Client: OAuth2Client, accessToken: String, idToken: Jwt)
```

Comment:
Maybe just name it "client"?

```suggestion
    suspend fun validate(client: OAuth2Client, accessToken: String, idToken: Jwt)
```